geopandas/base.py

```diff
@@ -144,22 +172,26 @@ def is_ring(self):
     @property
     def boundary(self):
         """Return the bounding geometry for each geometry"""
-        return _geo_unary_op(self, 'boundary')
+        x = vectorized.geo_unary_op('boundary', self._geometry_array.data)
+        return gpd.GeoSeries(x, index=self.index, crs=self.crs)
 
     @property
     def centroid(self):
         """Return the centroid of each geometry in the GeoSeries"""
-        return _geo_unary_op(self, 'centroid')
+        x = vectorized.geo_unary_op('centroid', self._geometry_array.data)
+        return gpd.GeoSeries(x, index=self.index, crs=self.crs)
```

Comment:
There is also a `unary_geo` defined above which could be used here? (since it is a small one, maybe not that worth of replacing those two lines, on the other hand it is more in line with the other ops)